src/Native/System.Native/pal_networking.cpp

```diff
@@ -2678,3 +2679,63 @@ extern "C" int32_t SystemNative_PlatformSupportsDualModeIPv4PacketInfo()
     return 0;
 #endif
 }
+
+static char* GetNameFromUid(uid_t uid)
+{
+    size_t bufferLength = 512;
+    while (true)
+    {
+        char *buffer = reinterpret_cast<char*>(malloc(bufferLength));
+        if (buffer == nullptr)
+            return nullptr;
+
+        struct passwd pw;
+        struct passwd* result;
+        if (getpwuid_r(uid, &pw, buffer, bufferLength, &result) == 0)
```

Comment:
It's a bit of a strangely designed method.  result is set to &pw on success or to NULL if the entry couldn't be found.  Either way, nothing to be freed.